x-pack/plugins/infra/public/utils/formatters/data.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { InfraWaffleMapDataFormat } from '../../lib/lib';
+import { formatNumber } from './number';
+
+const getPowerFromValue = (val: number, base: number, labels: string[]) => {
```

Comment:
Is it possible that this just brute-forces `Math.floor(Math.log(val) / Math.log(base))` (up to `labels.length`)?

Reply:
Ha Ha... So I was going to do that but I read that trick on Stack Overflow and wasn't sure how if that was plagiarizing or not so I just created my own. Is that a well known technique?

https://stackoverflow.com/questions/15900485/correct-way-to-convert-size-in-bytes-to-kb-mb-gb-in-javascript

Reply:
It is called "calculating the logarithm of `val` to the base `base`"... at least that what I learned in school :wink: